src/server/pps/server/api_server.go

```diff
@@ -2353,17 +2346,18 @@ func (a *apiServer) inspectPipelineInTransaction(txnCtx *txncontext.TransactionC
 				return nil, err
 			}
 		} else {
-			pipelineInfo.Service.IP = service.Spec.ClusterIP
+			pipelineInfo.Details.Service.IP = service.Spec.ClusterIP
 		}
 	}
 
+	// TODO: move this into
```

Comment:
helper function?

Reply:
The location that this code should be moved to is left as an exercise for the reader.